invenio/modules/workflows/static/js/workflows/selection.js

```diff
@@ -42,24 +42,24 @@ define(
     function HoldingPenSelection() {
 
       this.selection = function (ev, data) {
-        console.log("Selected:");
+        var selectedID = this.$node.dataTable().fnGetData(data)[1];
+        console.log("Selected: " + selectedID);
 
-        // row selected
-        console.log(data);
-
-        // current node this component is attached to
-        console.log(this.$node);
-
-        // using current node, get datatable instance and get data based on row
+        localStorage.addArrayItem('batchActionArray', selectedID)
```

Comment:
I am wondering for simplicity that we can instead store the array of IDs in a flight component attribute instead. For example:

```javascript
function HoldingPenSelection() {
      // ....

      this.attributes({
        selected_objects: []
      });

      // ....
```
Then you can access this attribute via `this.attr.selected_objects`. I think that would avoid using localStorage overall and simplify the code overall. Then when the array needs to be reset, it can be done via events - following the existing event based design.